Lib/distutils/command/check.py

```diff
@@ -97,15 +100,15 @@ def check_metadata(self):
         if metadata.author:
             if not metadata.author_email:
                 self.warn("missing meta-data: if 'author' supplied, " +
-                          "'author_email' must be supplied too")
+                          "'author_email' should be supplied too")
```

Comment:
This is incorrect: the `must` is guarded by the `if` here, so it’s fine

Reply:
Thank you for your input.

But as I wrote above in the pr description, the spec does not enforce a `author_email` field.
https://packaging.python.org/specifications/core-metadata/#author

So the word `must` is obviously wrong.

As a `proof of concept` I published a test package to PyPi (with author, without author_email) and it looks perfectly fine.

https://test.pypi.org/project/Products.ZopeTree/2.0.2.dev0/

Reply:
I don’t think you understood my message :)

The warning is not saying: you must always supply author-email
but: if you have author, you must have author-email

Hence, not incorrect!

Reply:
The message "if you have an author, you must have author-email" is **wrong** - this is the single reason for this pull request.

The specs do not require you to provide an email address even when you supply a name - only this more than 11 year old piece of code does for unknown and - when you compare it to the spec - wrong reasons.

So, I have a use case for "this is the author, but I do not know his email address" - and this is valid - if you follow the specs, but if you have a look in the code it does say "I must supply an email address".

Reply:
@merwok Yeah, the issue is that AFAICT from the spec, none of these fields are required *at all*. It's not clear why the warning about author/e-mail pairs was in there in the first place.

Reply:
Ah, I had misunderstood the situation!  Thanks for your patience.

The checks date from the first commit in 2003: https://github.com/python/cpython/commit/51a6a4c8354f413443253e73b4b41ecdca5c35f1 (later moved to a distinct `check` command)

That commit implements PEP 301, which does not mention author/maintainer fields (not even in the example setup.py script).  Looking at chronology, the active metadata PEP at the time was PEP 214, where author is optional but author-email is not! https://www.python.org/dev/peps/pep-0241/#author-email  The following version, PEP 314, has the same requirement.  This shows that the people who developed the original PyPI specs wanted each release to have an author email, and only PEP 345 (in 2005) made the fields optional.

This history doesn’t explain the distutils warnings; I can only guess that the author (@r1chardj0n3s) wanted to provide some guidance to developers, who can be expected to add their name to the metadata and forget to include the (at the time) required email field.